src/main/java/anab/Person/Trainer.java

```diff
@@ -0,0 +1,105 @@
+        this.specialization = specialization;
+        this.experienceYears = experienceYears;
+        this.feedbackMarks = feedbackMarks;
+    }
+
+    // see the specialization of a trainer
+    public String getSpecialization() {
+        return this.specialization;
+    }
+
+    //add specialization of a trainer
+    public void setSpecialization(String specialization) {
+        this.specialization = specialization;
+    }
+
+    //see the years of experience of a trainer
+    public int getExperience() {
+        return this.experienceYears;
+    }
+
+    //Add years of experience for a trainer
+    public void addExperience(String spec, int expYears) {
+        specialization = spec;
+        experienceYears = expYears;
+    }
+
+    public int setExperienceYears(int years) {
+        if (years != 0 && years > 0 && years >= this.experienceYears) {
+            years = getExperience() + years;
+            System.out.println("Trainer has " + years + " years of experience on " + this.specialization);
+
+        } else {
+            System.out.println("Add valid value for years of experience= ");
+        }
+        return years;
+    }
+
+    public void addFeedbackMark(int feedback) {
+        if (feedback > 0 && feedback <= 10) {
+            System.out.println("Student: anonymous, feedback note= " + feedback + ",specializ =  " + this.specialization);
+        } else {
+            System.out.println("Give a feedback mark from 0 to 10 ");
+        }
+    }
+
+    //Overloading method
+    public void addFeedbackMark(String name, int feedback) {
+        if (!name.isEmpty() && (feedback > 0 && feedback <= 10)) {
+            System.out.println("Student: " + name + ", feedback note= " + feedback + ",specializ =  " + this.specialization);
+        } else {
+            System.out.println("Give a feedback mark from 0 to 10 ");
+        }
+        this.feedbackMarks.put(name, feedback);
+    }
+
+    public void printFeedbackMarks() {
+        Map<String, Integer> feedbackMarks = new HashMap<String, Integer>();
+        Set set = feedbackMarks.entrySet();
+        Iterator itr = set.iterator();
+        while (itr.hasNext()) {
+            Map.Entry entry = (Map.Entry) itr.next();
+            System.out.println("Test print printFeedbackMarks: " + entry.getKey() + " " + entry.getValue());
+        }
+    }
+
+    public Map<String, Integer> getFeedbackMarks() {
+        return this.feedbackMarks;
+    }
+
+    @Override
+    public void printPersonDetails() {
+        super.printPersonDetails();
+        System.out.println("Specialization: " + specialization);
+        //  System.out.println("Experienced years: " + experienceYears);
+        System.out.println("Trainer feedback marks received = " + feedbackMarks);
+    }
+
+    public double averageFeedback() {
+        Map<String, Integer> feedbackMarks1 = new HashMap<String, Integer>();
+        int sum = 0;
+        Set set = feedbackMarks.entrySet();
```

Comment:
for calculating the average of the feedback marks, you should use the values from `feedbackMarks`

Reply:
feedbackMarks1 changed into feedbackMarks